site/src/lib/model/user.model.ts

```diff
@@ -361,6 +362,19 @@ export class User {
     return emailVerificationCode;
   }
 
+  async generateApiKey(db: Db, params: { displayName: string }) {
+    const { displayName } = params;
+
+    /* @todo allow users to have multiple API keys - remove this once implemented */
+    await ApiKey.revokeAll(db, { user: this });
+
+    return await ApiKey.create(db, { displayName, user: this });
```

Comment:
This is likely best addressed on the frontend, but I don't think it's expected behaviour for a "Create API Key" button to immediately invalidate an existing API key. This might lead a user to accidentally invalidate an API key they are using in some infrastructure. Until we remove this limitation, it might be a good idea to have some warning dialog on the FE to ensure the user is aware of the consequences of generating another API key, or just not let them generate one if they already have one.

Reply:
Agreed - the button will be called 'Regenerate API key' and a warning displayed when the user does so.